src/Concretize.hs

```diff
@@ -327,7 +327,7 @@ mkLambda visited allowAmbig _ tenv env root@(ListPat (FnPat fn arr@(ArrPat args)
                     modify (deleterDeps ++)
                     modify (copyFn :)
                     modify (copyDeps ++)
-            pure (Right [XObj (Fn (Just lambdaPath) (Set.fromList capturedVars)) (xobjInfo fn) (xobjTy fn), arr, recBody])
+            pure (Right [XObj (Fn (Just lambdaPath) (Set.fromList capturedVars)) (xobjInfo fn) (xobjTy fn), arr, body])
```

Comment:
Was the change from `recBody` to `body` intentional? I think this was needed to appropriately handle recursive anonymous functions in let bindings, e.g. `(let [f (fn [x] (if (= x 0) x (f (- x 1)))])`, since we rename the lambda (see renameRecursives above).  It doesn't seem like these changes would alter that behavior but maybe I'm missing something.